pkg/controller/common/helpers.go

```diff
@@ -539,12 +551,44 @@ func ValidateMachineConfig(cfg mcfgv1.MachineConfigSpec) error {
 	return nil
 }
 
+// ExtractIgnitionKargsFor4_13 parses a raw ignition config to 3.4 and extracts the kernel args,
+// erroring if ShouldNotExist is populated (since this MCO doesn't know how to implement it). This
+// function is used by the render controller to extract the kargs from ignition and migrate them to
+// MachineConfig so newer versions that specify kargs in ignition will still work during a downgrade.
+// TODO(jkyros): remove this when we raise the ignition default to 3.4
+func ExtractIgnitionKargsFor4_13(rawIgn []byte) ([]ign3_4types.KernelArgument, error) {
+	ignCfgV3, rptV3, errV3 := ign3_4.ParseCompatibleVersion(rawIgn)
+	// No kargs in an empty config
+	if errors.Is(errV3, ign3errors.ErrEmpty) {
+		return nil, nil
+	}
+	if errV3 == nil && !rptV3.IsFatal() {
```

Comment:
`errV3` shouldn't be nil if the report is fatal.  (And if it were, we'd silently return `nil, nil` from this function.)